### PR TITLE
[MINOR][DOCS] Update doc  `sql/README.md`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to update doc `sql/README.md`.

### Why are the changes needed?
After https://github.com/apache/spark/pull/41426, We have added a subproject `API` to our `SQL moudle`, so we need to update the doc `sql/README.md` synchronously.

### Does this PR introduce _any_ user-facing change?
Yes, make the doc clearer and more accurate.

### How was this patch tested?
Manually test.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: docs/sql-ref-syntax-aux-show-views.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ SHOW VIEWS FROM default LIKE 'sam*';
 | default   | sam1       | false        |
 +-----------+------------+--------------+
 
--- List all views from the current database matching the pattern `sam|suj｜temp*`
+-- List all views from the current database matching the pattern `sam|suj|temp*`
```
Comment (panbingkun) [inline docs/sql-ref-syntax-aux-show-views.md:102]: Not related to this PR, but I would like to take this opportunity to bring up this `correction`. (Full angle to half angle)

**Comment by panbingkun** [comment]
cc @amaliujia @HyukjinKwon

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

File: sql/README.md [hunk 2]
```diff
@@ -3,7 +3,8 @@ Spark SQL
 
 This module provides support for executing relational queries expressed in either SQL or the DataFrame/Dataset API.
 
-Spark SQL is broken up into four subprojects:
+Spark SQL is broken up into five subprojects:
+ - API (sql/api) - Includes some public API like DataType, Row, etc. This component can be shared between Catalyst and Spark Connect client.
```
Comment (amaliujia) [inline sql/README.md:7]: Thanks for the documentation!

**Review by amaliujia — COMMENTED** [review]
Late LGTM

**Pull request closed without merging**